src/FixedDataTableRow.js

```diff
@@ -129,6 +141,19 @@ class FixedDataTableRowImpl extends React.Component {
     onColumnReorderEnd: PropTypes.func,
   };
 
+  componentWillUpdate(nextProps, nextState){
+    if(this.waitingForMouseLeave &&
+      this.props.index !== nextProps.index){
+        /**
+         * This is necessary so that onMouseLeave is fired for the initial
+         * rowIndex and not for the one this row is replaced with.
+         */
+        if(this.mouseLeaveIndex === -1){
```

Comment:
Is there any advantage to updating this hear vs just setting it in onMouseEnter?  Setting in onMouseEnter seems cleaner to me.

Reply:
You are right, the solution had gotten unnecessarily complicated. That is much better and allows to simplify things quite a bit. I changed it to set the mouseLeaveIndex always onMouseEnter and always call onMouseLeave with the mouseLeaveIndex variable.